src/App.tsx

```diff
@@ -26,70 +27,128 @@ type ReorderOptions = {
-  console.log(sortType, isReversed);
+  visibleGoods.sort((good, nextGood) => {
+    switch (sortType) {
+      case SortType.ALPABET:
+        return good.localeCompare(nextGood);
+
+      case SortType.LENGTH:
+        return good.length - nextGood.length;
+
+      case SortType.NONE:
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+interface State {
+  isReversed: boolean;
+  sortType: SortType;
+}
+
+export class App extends React.Component<{}, State > {
+  state: Readonly<State> = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  sortByAlphabet = () => {
+    this.setState({
+      sortType: SortType.ALPABET,
+    });
+  };
+
+  sortByLength = () => {
+    this.setState({
+      sortType: SortType.LENGTH,
+    });
+  };
+
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  reset = () => {
+    this.setState({
+      isReversed: false,
+      sortType: SortType.NONE,
+    });
+  };
+
+  render(): React.ReactNode {
+    const { isReversed, sortType } = this.state;
+    const goods = getReorderedGoods(goodsFromServer, this.state);
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            onClick={this.sortByAlphabet}
+            className={classNames('button is-info',
+              { 'is-light': sortType !== SortType.ALPABET })}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            onClick={this.sortByLength}
+            className={classNames('button is-success',
+              { 'is-light': sortType !== SortType.LENGTH })}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            onClick={this.reverse}
+            className={classNames('button is-warning',
+              { 'is-light': !isReversed })}
+          >
+            Reverse
+          </button>
+
+          {(isReversed || sortType !== SortType.NONE)
```

Comment:
better to move such condition in variables
like `shouldBeResetButtonVisible` and use it here